core/src/main/java/org/apache/iceberg/avro/AvroFileAppender.java

```diff
@@ -40,7 +40,7 @@
   AvroFileAppender(Schema schema, OutputFile file,
                    Function<Schema, DatumWriter<?>> createWriterFunc,
                    CodecFactory codec, Map<String, String> metadata) throws IOException {
-    this.stream = file.create();
+    this.stream = file.createOrOverwrite();
```

Comment:
This will affect writes for all data and metadata files. Is it safe to do this everywhere?

Reply:
Moved it to Avro class and made configurable.